ArduPlane/ArduPlane.cpp

```diff
@@ -732,25 +732,43 @@ bool Plane::set_target_location(const Location& target_loc)
 // set target location (for use by scripting)
 bool Plane::get_target_location(Location& target_loc)
 {
-    switch (control_mode->mode_number()) {
-    case Mode::Number::RTL:
-    case Mode::Number::AVOID_ADSB:
-    case Mode::Number::GUIDED:
-    case Mode::Number::AUTO:
-    case Mode::Number::LOITER:
+    target_loc = next_WP_loc;
+    return true;
+}
+
+/*
+  update_target_location() works in all auto navigation modes
+ */
+bool Plane::update_target_location(const Location &old_loc, const Location &new_loc)
+{
+    if (!old_loc.same_latlon_as(next_WP_loc)) {
+        return false;
+    }
+    int32_t alt1, alt2;
+    if (!old_loc.get_alt_cm(Location::AltFrame::ABSOLUTE, alt1) ||
+        !next_WP_loc.get_alt_cm(Location::AltFrame::ABSOLUTE, alt2) ||
+        alt1 != alt2) {
+        return false;
+    }
+    next_WP_loc = new_loc;
+    next_WP_loc.change_alt_frame(old_loc.get_alt_frame());
+
+    return true;
+}
+
+// allow for velocity matching in VTOL
+bool Plane::set_velocity_match(const Vector2f &velocity)
```

Comment:
We do have direct access to quadplane from scripting, this could be a quadplane binding rather than a vehicle one.

Reply:
i'd like this for copter too in the future (leonard wants it for copter ship landing)